src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReaderProvider.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Internal;
+using System.Text;
+using System.Threading;
+
+namespace System.Reflection.Metadata
+{
+    public sealed class MetadataReaderProvider : IDisposable
+    {
+        // Either we have a provider and create metadata block lazily or
+        // we have no provider and metadata block is created in the ctor.
+        private MemoryBlockProvider _blockProviderOpt;
+        private AbstractMemoryBlock _lazyMetadataBlock;
+
+        private MetadataReaderProvider(AbstractMemoryBlock metadataBlock)
+        {
+            Debug.Assert(metadataBlock != null);
+            _lazyMetadataBlock = metadataBlock;
+        }
+
+        private MetadataReaderProvider(MemoryBlockProvider blockProvider)
+        {
+            Debug.Assert(blockProvider != null);
+            _blockProviderOpt = blockProvider;
+        }
+
+        /// <summary>
+        /// Creates a Portable PDB metadata provider over a blob stored in memory.
+        /// </summary>
+        /// <param name="start">Pointer to the start of the Portable PDB blob.</param>
+        /// <param name="size">The size of the Portable PDB blob.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="start"/> is <see cref="IntPtr.Zero"/>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="size"/> is negative.</exception>
+        /// <remarks>
+        /// The memory is owned by the caller and not released on disposal of the <see cref="MetadataReaderProvider"/>.
+        /// The caller is responsible for keeping the memory alive and unmodified throughout the lifetime of the <see cref="MetadataReaderProvider"/>.
+        /// The content of the blob is not read during the construction of the <see cref="MetadataReaderProvider"/>
+        /// </remarks>
+        public static unsafe MetadataReaderProvider FromPortablePdbImage(byte* start, int size) => FromMetadataImage(start, size);
```

Comment:
It's just convenience. We would need to actually read the metadata headers to check the version string to see if it's a Portable PDB metadata.